### ISSUE TITLE
stacking, named base models

### ISSUE THREAD

**Issue by bmreiniger** [rasbt/mlxtend#641]
It would be nice if the base models in the stacking ensembles explicitly required names, similar to sklearn's [ColumnTransformer](https://scikit-learn.org/stable/modules/generated/sklearn.compose.ColumnTransformer.html).  When all my base models are themselves pipelines, digging into the nest to find actual names becomes tedious.  (Especially in the context of set_params and grid searches, where the default behavior requires things like `pipeline-1__...`:  http://rasbt.github.io/mlxtend/user_guide/classifier/StackingCVClassifier/#example-3-stacked-cv-classification-and-gridsearch )

Maybe it's too late to try to change the API to _require_ names as in ColumnTransformer, but perhaps as an optional parameter?

**Comment by rasbt** [comment]
I agree that this can be helpful, especially for large sets of classifiers. I think having an optional way for naming them would be nice.
